src/librustc_lint/unused.rs

```diff
@@ -44,9 +44,11 @@ impl UnusedMut {
 
         let mut mutables = FxHashMap();
         for p in pats {
-            p.each_binding(|mode, id, _, path1| {
+            p.each_binding(|bm, id, _, path1| {
+                // FIXME(tschottdorf): this fires the assertion. Why?
+                // let bm = *cx.tables.pat_binding_modes.get(&p.id).expect("missing binding mode");
```

Comment:
Huh. That is indeed confusing; it would be helpful if you did something like this:

```
p.each_binding(|bm, id, span, path1| { // <-- I added `span` in place of `_`
  let bm = match cx.tables.pat_bindings_modes.get(&p.id) {
    Some(&bm) => bm,
    None => span_bug!(span, "missing binding mode")
  };
  ...
```